packages/fx-core/src/core/middleware/consolidateLocalRemote.ts

```diff
@@ -0,0 +1,326 @@
+      await fs.writeFile(manifestTemplatePath, JSON.stringify(manifest, null, 4));
+    }
+    fileList.push(
+      path.join(inputs.projectPath as string, "templates", "appPackage", "template.manifest.json")
+    );
+  } catch (e) {
+    for (const item of fileList) {
+      await fs.remove(item);
+    }
+    throw e;
+  }
+
+  // move old configs
+  const backupPath = path.join(inputs.projectPath as string, backupFolder);
+  sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateBackupConfigStart);
+  const localSettingsFile = path.join(
+    inputs.projectPath as string,
+    ".fx",
+    "configs",
+    "localSettings.json"
+  );
+  if (await fs.pathExists(localSettingsFile)) {
+    await fs.move(localSettingsFile, path.join(backupPath, ".fx", "configs", "localSettings.json"));
+  }
+  const localManifestFile = path.join(
+    inputs.projectPath as string,
+    "templates",
+    "appPackage",
+    "manifest.local.template.json"
+  );
+  if (await fs.pathExists(localManifestFile)) {
+    await fs.move(
+      localManifestFile,
+      path.join(backupPath, "templates", "appPackage", "manifest.local.template.json")
+    );
+  }
+  const remoteManifestFile = path.join(
+    inputs.projectPath as string,
+    "templates",
+    "appPackage",
+    "manifest.remote.template.json"
+  );
+  if (await fs.pathExists(remoteManifestFile)) {
+    await fs.move(
+      remoteManifestFile,
+      path.join(backupPath, "templates", "appPackage", "manifest.remote.template.json")
+    );
+  }
+
+  sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateBackupConfig);
+  sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateUpgrade);
+
+  postConsolidate(inputs.projectPath as string, ctx, inputs, backupFolder);
+  return true;
+}
+
+function checkMethod(ctx: CoreHookContext): boolean {
+  const methods: Set<string> = new Set(["getProjectConfig", "checkPermission"]);
+  if (ctx.method && methods.has(ctx.method) && fromReloadFlag) return false;
+  fromReloadFlag = ctx.method != undefined && methods.has(ctx.method);
+  return true;
+}
+
+async function postConsolidate(
+  projectPath: string,
+  ctx: CoreHookContext,
+  inputs: Inputs,
+  backupFolder: string | undefined
+): Promise<void> {
+  sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateGuideStart);
+  await updateGitIgnore(projectPath, TOOLS.logProvider, backupFolder);
+
+  TOOLS?.logProvider.warning(
+    `[core] Upgrade success! Old localSettings.json, manifest.local.template.json and manifest.remote.template.json have been backed up to the .backup folder and you can delete it.`
+  );
+
+  if (inputs.platform === Platform.VSCode) {
+    await globalStateUpdate(showConsolidateLogsFlag, true);
+    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateGuide, {
+      [TelemetryProperty.Status]: ProjectMigratorGuideStatus.Reload,
+    });
+    await TOOLS?.ui.reload?.();
```

Comment:
why do we need to reload the window after migration is done?

Reply:
Just to be consistent with upgrade scenario. Removed now.